File: object/Animal.py
import math
class Animal:
    def __init__(self,name, weightKg, heightM, isPredator, speedKph):
        self.name = name
        self.weightKg = weightKg
        self.heightM = heightM
        self.isPredator = isPredator
        self.speedKph = speedKph
        self.activityMultipller = 1.2
        
    def getBmi(self):
        return round(self.weightKg / (self.heightM ** 2),2)
    
    def getDailyCalories(self):
        return (math.floor((70 * self.weightKg ** 0.75) * self.activityMultipller * 100) / 100)
    

       

    def isDangerous(self):
        if self.isPredator or self.heightM >= 1.7 or self.speedKph >= 35:
            return True
        else:return False



rabbit = Animal("rabbit", 10, 0.3, False, 20)
print(rabbit.getBmi())
print(rabbit.isDangerous())
snake =  Animal("snake", 30, 1, True, 30)
print(snake.isDangerous())
print(snake.getDailyCalories())
elephant =  Animal("elephant", 300, 3, False, 5)
print(elephant.getBmi())
print(elephant.getDailyCalories())
gazelle =  Animal("gazelle", 50, 1.5, False, 100)
print(gazelle.getDailyCalories())
print(gazelle.isDangerous())

File: object/Animal.java
class Animal{
    public String name;
    public double speedKph;
    public double activityMultiplier = 1.2;
    public double weightKg;
    public double heightM;
    public boolean isPredator;

    public Animal(String name, double weightKg, double heightM, boolean isPredator, double speedKph){
        this.name = name;
        this.weightKg = weightKg;
        this.heightM = heightM;
        this.isPredator = isPredator;
        this.speedKph = speedKph;

    }

    public double getBmi(){
        return Math.floor(this.weightKg / Math.pow(this.heightM, 2) * 100) / 100;
    }

    public double getDailyCalories(){
        return Math.floor(70 * Math.pow(this.weightKg, 0.75) * activityMultiplier * 100) / 100;
    }


    public boolean isDangerous(){
        if(isPredator ||  this.speedKph >= 35 || this.heightM >= 170){
            return true;
    }
        else{
            return false;
    }
    }
}



class MyClass{
    public static void main(String[] args){
        Animal rabbit = new Animal("rabbit", 10, 0.3, false, 20);
        System.out.println(rabbit.getBmi());
        System.out.println(rabbit.isDangerous());

        Animal snake = new Animal("snake", 30, 1, true, 30);
        System.out.println(snake.isDangerous());
        System.out.println(snake.getDailyCalories());

        Animal elephant = new Animal("elephant", 300, 3, false, 5);
        System.out.println(elephant.getBmi());
        System.out.println(elephant.getDailyCalories());

        Animal gazelle = new Animal("gazelle", 50, 1.5, false, 100);
        System.out.println(gazelle.getDailyCalories());
        System.out.println(gazelle.isDangerous());

    }
}
